src/librbd/BlockGuard.h

```diff
@@ -109,10 +110,10 @@ class BlockGuard {
     ceph_assert(cell != nullptr);
     auto &detained_block_extent = reinterpret_cast<DetainedBlockExtent &>(
       *cell);
-    ldout(m_cct, 20) << "block_start="
+    ldout(m_cct, 20) << "[block_start="
                      << detained_block_extent.block_extent.block_start << ", "
                      << "block_end="
-                     << detained_block_extent.block_extent.block_end << ", "
+                     << detained_block_extent.block_extent.block_end << "), "
```

Comment:
you need to update `DetainedBlockExtentCompare::operator()` accordingly.

Reply:
@tchaikov Do you mean that we should use "<" instead of "<=" in it?

Reply:
i am copying the related code here for reference:

```c++
  struct DetainedBlockExtentCompare {
    bool operator()(const BlockExtent &lhs,
                    const BlockExtent &rhs) const {
      // check for range overlap (lhs < rhs)
      if (lhs.block_end <= rhs.block_start) {
        return true;
      }
      return false;
    }
  };
```
after a second thought, i think the existing code is wrong actually. as per https://www.boost.org/doc/libs/1_72_0/doc/html/intrusive/set_multiset.html#intrusive.set_multiset.set_multiset_containers , 

> The comparison functor must induce a strict weak ordering

so, the comparator of "f" should fulfill the requirements of https://en.wikipedia.org/wiki/Weak_ordering#Strict_weak_orderings . in other words, it should look like "<". but, since before this change, "BlockExtent" is inclusive, as "range_a" of [1, 1]" satisfies range_a < range_a, because "1 <= 1".

please note, i am not trying to give an answer. just to put that the code looks suspicious.

Reply:
@tchaikov 
I got your point. Whatever inclusive or exclusive, there's one corner case in the current implementation: block_start equals to block_end and use another same range BlockExtent to do compare.

Reply:
Except it's technically an invalid case, right? [1, 1) is a length zero interval which shouldn't be allowed.